samples/extensions/extended_dynamic_state2/README.md

```diff
@@ -0,0 +1,387 @@
+	graphics_create.pViewportState      = &viewport_state;
+	graphics_create.pDepthStencilState  = &depth_stencil_state;
+	graphics_create.pDynamicState       = &dynamic_state;
+	graphics_create.pVertexInputState   = &vertex_input_state;
+	graphics_create.pTessellationState  = VK_NULL_HANDLE;
+	graphics_create.stageCount          = 2;
+	graphics_create.pStages             = shader_stages.data();
+	graphics_create.layout              = pipeline_layouts.baseline;
+
+	
+
+	graphics_create.pNext      = VK_NULL_HANDLE;
+	graphics_create.renderPass = render_pass;
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline.baseline));
+...
+```
+
+And now, thanks to dynamic state 2, we can change parameters before each corresponding draw call.
+
+```C++
+	vkCmdBindDescriptorSets(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layouts.baseline, 0, 1, &descriptor_sets.baseline, 0, nullptr);
+	vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline.baseline);
+
+	vkCmdSetPrimitiveTopologyEXT(draw_cmd_buffers[i], VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST);
+	vkCmdSetPrimitiveRestartEnableEXT(draw_cmd_buffers[i], VK_FALSE);
+	draw_from_scene(draw_cmd_buffers[i], &scene_nodes, SCENE_BASELINE_OBJ_INDEX);
+
+	vkCmdSetDepthBiasEnableEXT(draw_cmd_buffers[i], VK_FALSE);
+	vkCmdSetRasterizerDiscardEnableEXT(draw_cmd_buffers[i], VK_FALSE);
+	vkCmdSetPrimitiveTopologyEXT(draw_cmd_buffers[i], VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP);
+	vkCmdSetPrimitiveRestartEnableEXT(draw_cmd_buffers[i], VK_TRUE);
+
+	draw_created_model(draw_cmd_buffers[i]);
+
+	vkCmdBindDescriptorSets(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layouts.tesselation, 0, 1, &descriptor_sets.tesselation, 0, nullptr);
+	vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline.tesselation);
+
+	vkCmdSetPrimitiveTopologyEXT(draw_cmd_buffers[i], VK_PRIMITIVE_TOPOLOGY_PATCH_LIST);
+	vkCmdSetPatchControlPointsEXT(draw_cmd_buffers[i], gui_settings.patch_control_points);
+
+	draw_from_scene(draw_cmd_buffers[i], &scene_nodes, SCENE_TESSELLATION_OBJ_INDEX);
+
+	vkCmdBindDescriptorSets(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layouts.background, 0, 1, &descriptor_sets.background, 0, nullptr);
+	vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline.background);
+
+	draw_model(background_model, draw_cmd_buffers[i]);
+
+```
+
+
+
+## Enabling the Extension
+The extended dynamic state 2 api is provided in Vulkan 1.3 and the appropriate headers / SDK is required.
+
+The device extension is provided by `VK_EXT_EXTENDED_DYNAMIC_STATE_2_EXTENSION_NAME`. It also requires 
+`VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME` instance extension to be enabled
+
+```C++
+add_instance_extension(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+add_device_extension(VK_EXT_EXTENDED_DYNAMIC_STATE_2_EXTENSION_NAME);
```

Comment:
Add code and description for enabling particular features from extension. (like in the description from vertex input dynamic state)

Reply:
Done